### ISSUE TITLE
[Dev support]: What is the correct way of dealing with file uploads using route handlers?

### ISSUE THREAD

**Issue by donatas-xyz** [microsoft/teams-ai#2226]
## Question
I may be missing something obvious, but after trying a few things and checking .NET [samples ](https://github.com/microsoft/teams-ai/blob/d3356b6ab4258067493a9a617cad782cb1667148/dotnet/samples/06.assistants.b.orderBot/Program.cs) with `TeamsAttachmentDownloader`, I don't seem to be able to progress with the conversation flow, if message has no attachment? 

## Code snippets
Say we have our typical handler for capturing all messages (so far it looks like the only way to get attachments?): 
```csharp 
// Listen for ANY message to be received. MUST BE AFTER ANY OTHER MESSAGE HANDLERS
app.OnActivity(ActivityTypes.Message, ActivityHandlers.MessageActivityHandler);
```

Then in `ActivityHandlers` class we have something like: 
```csharp
public RouteHandler<AppState> MessageActivityHandler => async (turnContext, turnState, cancellationToken) =>
{
    // Check if the activity contains attachments
    if (turnContext.Activity.Attachments != null && turnContext.Activity.Attachments.Any())
    {
        // Attachment handling logic
    }
    else 
    {
         // What is supposed to go here to continue with the conversation flow, when message has no attachment?? 
    }
}
```

There are [examples](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-file-upload/csharp/Bots/TeamsFileUploadBot.cs) that use `TeamsActivityHandler`, but I believe `RouteHandler` has replaced it? 

Thank you!

**Comment by corinagum** [comment]
[Activity handlers](https://github.com/microsoft/teams-ai/blob/d3356b6ab4258067493a9a617cad782cb1667148/getting-started/MIGRATION/02.DOTNET.md#activity-handler-methods)

Please see the [CardGazer Bot](https://github.com/microsoft/teams-ai/blob/d3356b6ab4258067493a9a617cad782cb1667148/dotnet/samples/04.ai.f.vision.cardMaster/CardGazerBot.cs) for an AI scenario with attachments.

**Comment by donatas-xyz** [comment]
Hi @corinagum 

I'm sorry, but there must of been a misunderstanding - I don't have issues with actions - it's the handlers I'm after. The **CardGazer Bot** is only focused on actions, and those are mostly (always?) intentional, therefore easier to control the logic flow, because there is a clear entrance point and a result expected at the end. 

My issue, however is with the scenario, where user simply throws a file in to the Teams chat window - what should happen then? Your **Activity handlers** link provides an insight of what should happen next, which was useful to know - a consent card should be shown to user to save the file and then it can progress from there - again, this would result in a clear goal and a result (user's file either gets saved or not). 

But what about if message has no attachment? Since `OnActivity` is handling all types of messages I should be able to continue with the conversation, however currently the conversation simply stops. The examples that I found are only echoing user's requests, which is nowhere near being helpful as this doesn't replicate any real world usage. 

Perhaps I need to rethink my approach to user file uploads, but the use case would still stand - what if user just randomly throws a file into the chat window? 

I've tried to apply CardGazer Bot's logic in my app, but LLM doesn't seem to understand the concept of image/file being loaded into the chat window without using `app.OnActivity(ActivityTypes.Message, ActivityHandlers.MessageActivityHandler);`. I'm using GPT-4 model, but an older API,. so the `tools` augmentation is not supported in my case. Is that a requirement for this example to work? 

If I get it to work - would that be the correct way of handling file uploads by a user - a file should always be accompanied by an instruction on what to do with the file, which would then lead to a predefined action? This sounds reasonable for what I want to achieve, but still - in my initial example I still can't see the way to continue with the conversation if none of the "checkpoints" have been hit? 

Thank you for reading all this.
